### PR TITLE
Move docs to npm workspaces

### PR DESCRIPTION
**Pull request by himadrisingh**
deploy to netlify using github actions

### CODE REVIEW

**Comment by begelundmuller** [comment]
Love to see this cleanup. One question: why deploy to Netlify using GH actions instead of the current webhooks-based solution (where things are configured in the Netlify control plane or in a `netlify.toml` file)?

**Comment by begelundmuller** [comment]
One other note: it would be nice to have a `docs-check` Github Action that checks docs compilation on PRs that edit docs. We had a problem [here](https://rilldata.slack.com/archives/CTZ8XBQ85/p1665764391974059) where someone edited the docs and didn't realize they used a bad link.

**Review by begelundmuller — CHANGES_REQUESTED** [review]
Following up on the note [here](https://rilldata.slack.com/archives/C02T907FEUB/p1669658781699309?thread_ts=1669642759.226309&cid=C02T907FEUB), let's just keep deploying docs with Netlify from `main` for now. So I think we can remove the docs action.

Also, this still needs the docs checks actions for checking docs changes in PRs

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Build and Deploy to Netlify
+on:
+  push:
+    tags:
+      - "*"
```
Comment (himadrisingh) [inline .github/workflows/docs.yml:5]:
```suggestion
    branches: [ "main" ]
```

**Review by begelundmuller — COMMENTED** [review]
1. Can we just use Netlify's webhooks-based deploys that we used before and remove the `docs.yml` fully? If we ever add docs versioning, I think it'll be using Docusaurus versioning, which will still require us to deploy continuously from main.

2. It's missing a `docs-check.yml` workflow for checking that docs changes compile succesfully on PRs (see comment [here](https://github.com/rilldata/rill-developer/pull/1307#issuecomment-1328859111))

**Review by begelundmuller — APPROVED** [review]
LGTM

**Pull request merged**